### PR TITLE
infra: use typescript project references

### PR DESCRIPTION
**Pull request by AviVahl**
- fixes cross-package .d.ts generation. typescript now emits proper package imports for inferred indirect types.
- separate dashboard to own package (@wixc3/engine-dashboard).
- move engineer's dev-server feature into ./src/feature (instance exported from package itself).
- remove tsconfig-paths and related configuration.
- ensure all test files use public api (!!!).
- expose template creation api from scripts.
- simplified each package scripts.
- all packages output to `./dist`. core's umd build is built to `./dist/umd`.
- increased file server ix timeout to combat flakiness.
- rename `engine` references to `engineer`.
- `skipLibCheck` is now on by default on `tsc --init`, so activate it.
- correct electron app example title.
- adjust ci scripts to `yarn build` before publishing. removed `prepack` scripts, except for core's umd build.

### CODE REVIEW

File: packages/engineer/bin/engineer.js [hunk 1]
```diff
@@ -1,17 +1,3 @@
...
-    require.extensions['.tsx'] = nodeExtension;
-
-    const { options: tsconfigPathsOptions } = require('tsconfig-paths/lib/options');
-    tsconfigPathsOptions.cwd = dirname(configFilePath);
-    require('tsconfig-paths/register');
-    require('../src/cli');
-} else {
-    require('../cjs/src/cli');
-}
+require('../dist/cli.js');
```
Comment (AdamYahid) [inline packages/engineer/bin/engineer.js:3]: doesn't this mean I have to build for every change I make?

Comment (AviVahl) [reply packages/engineer/bin/engineer.js:3]: yes.

**Review by AdamYahid — APPROVED** [review]

File: packages/core-node/test/process-entry.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { Communication } from '@wixc3/engine-core';
-import { IPCHost } from '../src/ipc-host';
+import { IPCHost } from '@wixc3/engine-core-node';
```
Comment (RomanYarik) [inline packages/core-node/test/process-entry.ts:2]: why? you are inside that package

File: packages/core/src/com/hosts/event-emitter.ts [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
 export class EventEmitterHost extends BaseHost {
-    constructor(private host: EventEmitter) {
+    constructor(private host: EventEmitter<{ message: Message }>) {
         super();
 
         this.host.on('message', (data) => this.emitMessageHandlers(data));
     }
 
-    public postMessage(message: any) {
+    public postMessage(message: Message) {
```
Comment (RomanYarik) [inline packages/core/src/com/hosts/event-emitter.ts:12]: :+1:

**Review by RomanYarik — APPROVED** [review]

**Pull request merged**